synapse/storage/databases/state/store.py

```diff
@@ -404,6 +406,126 @@ def _insert_into_cache(
                 fetched_keys=non_member_types,
             )
 
+    async def store_state_deltas_for_batched(
+        self,
+        events_and_context: List[Tuple[EventBase, UnpersistedEventContextBase]],
+        room_id: str,
+        prev_group: int,
```

Comment:
Why do we need to pass this in? Can we not get it from the the first context's `prev_group`?

Reply:
We need this because we don't have really any state group info in the UnpersistedEventContext at this juncture - that info is all assigned/calculated in the `persist` or in this case the batch persist function of the class UnpersistedEventContex, which this function is a part of.